homeassistant/components/permobil/binary_sensor.py

```diff
@@ -0,0 +1,97 @@
+"""Platform for binary sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from mypermobil import BATTERY_CHARGING
+
+from homeassistant import config_entries
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import MyPermobilCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PermobilBinaryRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    is_on_fn: Callable[[Any], bool]
+    available_fn: Callable[[Any], bool]
+
+
+@dataclass(frozen=True)
+class PermobilBinarySensorEntityDescription(
+    BinarySensorEntityDescription, PermobilBinaryRequiredKeysMixin
+):
+    """Describes Permobil binary sensor entity."""
+
+
+BINARY_SENSOR_DESCRIPTIONS: tuple[PermobilBinarySensorEntityDescription, ...] = (
+    PermobilBinarySensorEntityDescription(
+        is_on_fn=lambda data: data.battery[BATTERY_CHARGING[0]],
+        available_fn=lambda data: BATTERY_CHARGING[0] in data.battery,
+        key="is_charging",
+        translation_key="is_charging",
+        icon="mdi:battery_unknown",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create and setup the binary sensor."""
+    coordinator: MyPermobilCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        PermobilbinarySensor(coordinator=coordinator, description=description)
+        for description in BINARY_SENSOR_DESCRIPTIONS
+    )
+
+
+class PermobilbinarySensor(
+    CoordinatorEntity[MyPermobilCoordinator], BinarySensorEntity
+):
```

Comment:
Does it maybe make sense for the integration to have a shared base entity and maybe a device?